inst/cpp/sim.cpp

```diff
@@ -63,23 +63,116 @@ void pk_code (int i, std::vector<double> times, std::vector<double> doses, doubl
   // insert custom pk event code
 }
 
```

Comment:
this file is becoming fairly large. In principle we should separate off into modular design. However, some issues with that, see [RXR-2712](https://insight-rx.atlassian.net/browse/RXR-2712). We should tackle in separate PR.